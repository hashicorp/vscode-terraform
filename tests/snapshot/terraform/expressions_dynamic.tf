resource "thing" "name" {
  name = "tf-test-name"

  dynamic "setting" {
    for_each = var.settings
    content {
      namespace = setting.value["namespace"]
      name = setting.value["name"]
      value = setting.value["value"]
    }
  }

  dynamic "origin_group" {
    for_each = var.load_balancer_origin_groups
    content {
      name = origin_group.key

      dynamic "origin" {
        for_each = origin_group.value.origins
        content {
          hostname = origin.value.hostname
        }
      }
    }
  }
}
