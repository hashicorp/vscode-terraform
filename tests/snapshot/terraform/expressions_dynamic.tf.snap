>resource "thing" "name" {
#^^^^^^^^ source.terraform meta.block.terraform entity.name.type.terraform
#        ^ source.terraform meta.block.terraform
#         ^ source.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.begin.terraform
#          ^^^^^ source.terraform meta.block.terraform entity.name.tag.terraform
#               ^ source.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.end.terraform
#                ^ source.terraform meta.block.terraform
#                 ^ source.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.begin.terraform
#                  ^^^^ source.terraform meta.block.terraform entity.name.tag.terraform
#                      ^ source.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.end.terraform
#                       ^ source.terraform meta.block.terraform
#                        ^ source.terraform meta.block.terraform punctuation.section.block.begin.terraform
>  name = "tf-test-name"
#^^ source.terraform meta.block.terraform
#  ^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#      ^ source.terraform meta.block.terraform variable.declaration.terraform
#       ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#        ^ source.terraform meta.block.terraform variable.declaration.terraform
#         ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#          ^^^^^^^^^^^^ source.terraform meta.block.terraform string.quoted.double.terraform
#                      ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
>
>  dynamic "setting" {
#^^ source.terraform meta.block.terraform
#  ^^^^^^^ source.terraform meta.block.terraform meta.block.terraform entity.name.label.terraform
#         ^ source.terraform meta.block.terraform meta.block.terraform
#          ^ source.terraform meta.block.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.begin.terraform
#           ^^^^^^^ source.terraform meta.block.terraform meta.block.terraform entity.name.tag.terraform
#                  ^ source.terraform meta.block.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.end.terraform
#                   ^ source.terraform meta.block.terraform meta.block.terraform
#                    ^ source.terraform meta.block.terraform meta.block.terraform punctuation.section.block.begin.terraform
>    for_each = var.settings
#^^^^ source.terraform meta.block.terraform meta.block.terraform
#    ^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#            ^ source.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#             ^ source.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#              ^ source.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#               ^^^ source.terraform meta.block.terraform meta.block.terraform support.constant.terraform
#                  ^ source.terraform meta.block.terraform meta.block.terraform keyword.operator.accessor.terraform
#                   ^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform variable.other.member.terraform
>    content {
#^^^^ source.terraform meta.block.terraform meta.block.terraform
#    ^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform entity.name.label.terraform
#           ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#            ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform punctuation.section.block.begin.terraform
>      namespace = setting.value["namespace"]
#^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#      ^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#               ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#                ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#                 ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#                  ^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#                         ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform keyword.operator.accessor.terraform
#                          ^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.other.member.terraform
#                               ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform punctuation.section.brackets.begin.terraform
#                                ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#                                 ^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform string.quoted.double.terraform
#                                          ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
#                                           ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform punctuation.section.brackets.end.terraform
>      name = setting.value["name"]
#^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#      ^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#          ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#           ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#            ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#             ^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#                    ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform keyword.operator.accessor.terraform
#                     ^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.other.member.terraform
#                          ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform punctuation.section.brackets.begin.terraform
#                           ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#                            ^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform string.quoted.double.terraform
#                                ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
#                                 ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform punctuation.section.brackets.end.terraform
>      value = setting.value["value"]
#^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#      ^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#           ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#            ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#             ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#              ^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#                     ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform keyword.operator.accessor.terraform
#                      ^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.other.member.terraform
#                           ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform punctuation.section.brackets.begin.terraform
#                            ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#                             ^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform string.quoted.double.terraform
#                                  ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
#                                   ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform punctuation.section.brackets.end.terraform
>    }
#^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#    ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform punctuation.section.block.end.terraform
>  }
#^^ source.terraform meta.block.terraform meta.block.terraform
#  ^ source.terraform meta.block.terraform meta.block.terraform punctuation.section.block.end.terraform
>
>  dynamic "origin_group" {
#^^ source.terraform meta.block.terraform
#  ^^^^^^^ source.terraform meta.block.terraform meta.block.terraform entity.name.label.terraform
#         ^ source.terraform meta.block.terraform meta.block.terraform
#          ^ source.terraform meta.block.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.begin.terraform
#           ^^^^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform entity.name.tag.terraform
#                       ^ source.terraform meta.block.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.end.terraform
#                        ^ source.terraform meta.block.terraform meta.block.terraform
#                         ^ source.terraform meta.block.terraform meta.block.terraform punctuation.section.block.begin.terraform
>    for_each = var.load_balancer_origin_groups
#^^^^ source.terraform meta.block.terraform meta.block.terraform
#    ^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#            ^ source.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#             ^ source.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#              ^ source.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#               ^^^ source.terraform meta.block.terraform meta.block.terraform support.constant.terraform
#                  ^ source.terraform meta.block.terraform meta.block.terraform keyword.operator.accessor.terraform
#                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform variable.other.member.terraform
>    content {
#^^^^ source.terraform meta.block.terraform meta.block.terraform
#    ^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform entity.name.label.terraform
#           ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#            ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform punctuation.section.block.begin.terraform
>      name = origin_group.key
#^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#      ^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#          ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#           ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#            ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#             ^^^^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#                         ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform keyword.operator.accessor.terraform
#                          ^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.other.member.terraform
>
>      dynamic "origin" {
#^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#      ^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform entity.name.label.terraform
#             ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#              ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.begin.terraform
#               ^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform entity.name.tag.terraform
#                     ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.end.terraform
#                      ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#                       ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform punctuation.section.block.begin.terraform
>        for_each = origin_group.value.origins
#^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#        ^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#                ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#                 ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#                  ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#                   ^^^^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#                               ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform keyword.operator.accessor.terraform
#                                ^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.other.member.terraform
#                                     ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform keyword.operator.accessor.terraform
#                                      ^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.other.member.terraform
>        content {
#^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#        ^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform entity.name.label.terraform
#               ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#                ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform punctuation.section.block.begin.terraform
>          hostname = origin.value.hostname
#^^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#          ^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#                  ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#                   ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#                    ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#                     ^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#                           ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform keyword.operator.accessor.terraform
#                            ^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.other.member.terraform
#                                 ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform keyword.operator.accessor.terraform
#                                  ^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform variable.other.member.terraform
>        }
#^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#        ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform punctuation.section.block.end.terraform
>      }
#^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#      ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform meta.block.terraform punctuation.section.block.end.terraform
>    }
#^^^^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform
#    ^ source.terraform meta.block.terraform meta.block.terraform meta.block.terraform punctuation.section.block.end.terraform
>  }
#^^ source.terraform meta.block.terraform meta.block.terraform
#  ^ source.terraform meta.block.terraform meta.block.terraform punctuation.section.block.end.terraform
>}
#^ source.terraform meta.block.terraform punctuation.section.block.end.terraform
>